src/python/pants/option/ranked_value.py

```diff
@@ -27,10 +27,10 @@ def __new__(cls, rank: int, display: str) -> Rank:
         member._rank = rank
         return member
 
-    def __lt__(self, other: Any) -> Union["NotImplemented", bool]:
```

Comment:
The return type went from correct to incorrect. What was the mypy complaint that forced lying about Not implemented not being returned?

Reply:
Oh, Ok - you removed the Not implemented bit. I read the diff badly. Why did you do that?

Reply:
```
"builtins.NotImplemented" is not valid as a type  [valid-type]
        def __lt__(self, other: Any) -> bool | NotImplemented:
```

Reply:
Warping code to "work" with a buggy type checker seems bad: https://github.com/python/mypy/issues/4791

I'd prefer a type ignore with a pointer to the issue unless you think this actually made the code better in some way.

Reply:
Hm we're not warping code, though. The function return type is ignored at runtime. The only runtime change is now using cast, but that's a no-op. 

I didn't spend much time trying to figure this out because it's annoying MyPy doesn't support it, but not a very big deal given that it's isolated to only this definition.

Reply:
Ok, you're right - not warping, the return Not implemented is still there. It's just a lie then. If another burn is needed for some other reason please consider documenting the typing lie with a link to the mypy issue.